wordpress-mobile/WordPress-iOS
# Menus: Readability fixes

Fixes #5683

Readability fixes for the labels in the header. This was a bit of tech debt, but also more troublesome with our new split-view on iPad and iPhone Pluses.

In the Menus header, refactored to use a `UIStackView` and allow for properly resizing/truncating the selection labels when it's super long. Also updated the storyboards for Xcode 8.2 and fixed the compression issue for the "USES" label.

To test:
1. Open Menus on an iPhone Plus simulator or device, for a test site that has at least one Menu set up on it.
2. Rotate to landscape, this will bring in the split-view and compress the layout.
3. Type in a new, long name for the Menu currently selected (may be the "Default Menu").
4. Check that the "USES" label doesn't compress in width when you reach the truncated limit.
5. Tap on the Menu selection dropdown at the top-right.
6. Check that the new Menu name truncates correctly before the checkmark or edge of the view.

Needs review: @bummytime can you take a peak at this one?
